beacon_node/beacon_chain/src/blob_verification.rs

```diff
@@ -374,16 +393,35 @@ pub fn validate_blob_sidecar_for_gossip<T: BeaconChainTypes>(
         return Err(GossipBlobError::InvalidInclusionProof);
     }
 
+    let fork_choice = chain.canonical_head.fork_choice_read_lock();
+
     // We have already verified that the blob is past finalization, so we can
     // just check fork choice for the block's parent.
-    let Some(parent_block) = chain
-        .canonical_head
-        .fork_choice_read_lock()
-        .get_block(&block_parent_root)
-    else {
+    let Some(parent_block) = fork_choice.get_block(&block_parent_root) else {
         return Err(GossipBlobError::BlobParentUnknown(blob_sidecar));
     };
 
+    // If fork choice does *not* consider the parent to be a descendant of the finalized block,
+    // then there are two more cases:
+    //
+    // 1. We have the parent stored in our database. Because fork-choice has confirmed the
+    //    parent is *not* in our post-finalization DAG, all other blocks must be either
+    //    pre-finalization or conflicting with finalization.
+    // 2. The parent is unknown to us, we probably want to download it since it might actually
+    //    descend from the finalized root.
+    if !fork_choice.is_finalized_checkpoint_or_descendant(block_parent_root) {
+        if chain
+            .store
+            .block_exists(&block_parent_root)
```

Comment:
I think this condition should be after the proposer signature verification.
Else it would be possible for an attacker to spam us with useless blobs with bogus non-canonical parent roots and make us do a bunch of database reads.